app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1235,7 +1239,12 @@ private void setMapMarkerActions(Marker selectedMarker) {
     @Override
     public void recenterMap(fr.free.nrw.commons.location.LatLng curLatLng) {
         if (curLatLng == null) {
+            showLocationNullDialog();
```

Comment:
There's an unintended consequence of handling the `null` state of `curLatLng` this way, I suppose. Consider this, the location is on and is in "High Accuracy mode" but the device has not yet been located. In this case tapping on the "Locate me" icon shows the the "Turn on location?" dialog each time I tap on it. So, it looks like we get `null` in `curLatLng` for more cases than [you mentioned](https://github.com/commons-app/apps-android-commons/issues/3397#issuecomment-591000521). This behaviour is not ideal and should be corrected.

Note that though the behaviour I describe is accurate, I'm _speculating_ that I'm observing that behaviour as a consequence of `curLatLng` being `null`. So, there's a possibility that I'm observing the behaviour for different reasons. Anyways, can you look into how to fix this?

Reply:
Hi @sivaraam!  I am actually busy working on a project and would most likely be able to continue the discussion and make the changes in a day or two. I apologize for the delay.

Reply:
> There's an unintended consequence of handling the `null` state of `curLatLng` this way, I suppose. Consider this, the location is on and is in "High Accuracy mode" but the device has not yet been located. In this case tapping on the "Locate me" icon shows the the "Turn on location?" dialog each time I tap on it. So, it looks like we get `null` in `curLatLng` for more cases than [you mentioned](https://github.com/commons-app/apps-android-commons/issues/3397#issuecomment-591000521). This app behaviour is not ideal and should be corrected.

I agree. I tried to reproduce this behaviour just now and I was able to do it by turning my internet off and location set to High Accuracy.

In order to deal with this NULL state, one choice is we don't show the Dialog when location is NULL ie when location is not available since we already get a toast for that. I think this actually makes sense since the Dialog is to prompt the user to enable location in case it's off and not when it's NULL. Any suggestions?

Reply:
> > There's an unintended consequence of handling the `null` state of `curLatLng` this way, I suppose. Consider this, the location is on and is in "High Accuracy mode" but the device has not yet been located. In this case tapping on the "Locate me" icon shows the the "Turn on location?" dialog each time I tap on it. So, it looks like we get `null` in `curLatLng` for more cases than [you mentioned](https://github.com/commons-app/apps-android-commons/issues/3397#issuecomment-591000521). This app behaviour is not ideal and should be corrected.
> 
> I agree. I tried to reproduce this behaviour just now and I was able to do it by turning my internet off and location set to High Accuracy.
> 
> In order to deal with this NULL state, one choice is we don't show the Dialog when location is NULL ie when location is not available since we already get a toast for that. I think this actually makes sense since the Dialog is to prompt the user to enable location in case it's off and not when it's NULL. Any suggestions?

We can check if the internet is enabled or not as a listener is slready set up in the app but I am afraid that this will be a messy way to handle this since the same conditions can occur due to different reasons which we might not be able to foresee now. What is your opinion @sivaraam ?

Reply:
Also, I know that this PR and its issue have been pending for too long now but I just want to make sure that we do this right because otherwise, it could cost us more time afterwards. So, thanks for the patience.

Reply:
> Any suggestions?

I'm sorry. I totally missed this comment. Feel free to ping me when I haven't responded within a reasonable amount of time.

> one choice is we don't show the Dialog when location is NULL ie when location is not available since we already get a toast for that. I think this actually makes sense since the Dialog is to prompt the user to enable location in case it's off and not when it's NULL

I'm still not sure in what all cases we get NULL here. Is there any case other than "location in device only mode" and "location is in high accuracy mode but device not yet been found precisely" for which we get NULL?

In contrast, if Nearby doesn't work in "device only mode" wouldn't it be better to show the dialog when location is set to "device only mode". Can you check if there is a reliable way to identify if the location is in "device only mode"? I suppose the Play Services API could help us here but [we decided not to use it](https://github.com/commons-app/apps-android-commons/issues/3435) so I think we would have to find another way, if available. Or we might have to re-consider and re-discuss about that decision. 

> We can check if the internet is enabled or not as a listener is slready set up in the app but I am afraid that this will be a messy way to handle this ...

Yeah, it's messy indeed.

> ... since the same conditions can occur due to different reasons which we might not be able to foresee now. 

Exactly. You observed this when you turned off the internet but when I observed this my device's internet was *not* turned off. It just took some time to get the accurate location of the device in my case (and this happens a lot for me). When I tap on the "Locate me" button in the meanwhile I see the pop-up.

> ... I just want to make sure that we do this right because otherwise, it could cost us more time afterwards.

Perfect! It's always better to do things right than to rush them and regret when it bites us later.

> So, thanks for the patience.

Actually, thanks a lot to you for patiently listening to feedback and improvising the PR! It's appreciated. 🙂

Reply:
> I'm sorry. I totally missed this comment. Feel free to ping me when I haven't responded within a reasonable amount of time.

I will keep that in mind :)

> Is there any case other than "location in device only mode" and "location is in high accuracy mode but device not yet been found precisely" for which we get NULL?

Other than the internet off case I mentioned above, I don't know. I didn't find any other case but I am sure there are more circumstances that can cause this.

Reply:
> Actually, thanks a lot to you for patiently listening to feedback and improvising the PR! It's appreciated.

I am just trying to learn from you guys as fast as possible! :nerd_face: